browser/lib/utils.js

```diff
@@ -15,43 +15,12 @@ export function escapeHtmlCharacters (text) {
     return str
   }
 
-  let escape
-  let html = ''
-  let index = 0
-  let lastIndex = 0
-
-  for (index = match.index; index < str.length; index++) {
-    switch (str.charCodeAt(index)) {
-      case 34: // "
-        escape = '&quot;'
-        break
-      case 38: // &
-        escape = '&amp;'
-        break
-      case 39: // '
-        escape = '&#39;'
-        break
-      case 60: // <
-        escape = '&lt;'
-        break
-      case 62: // >
-        escape = '&gt;'
-        break
-      default:
-        continue
-    }
-
-    if (lastIndex !== index) {
-      html += str.substring(lastIndex, index)
-    }
-
-    lastIndex = index + 1
-    html += escape
-  }
-
-  return lastIndex !== index
-      ? html + str.substring(lastIndex, index)
-      : html
+  return str
```

Comment:
Have you tested the speed between the current and new version of the function?

From my point of view, 6 replaces is the slower solution than previous. Instead of searching the whole text (which can be big) only once it is done 6 times in a row.

Reply:
Yes, I've noticed that and discovered that your version seem to work fine and not the source of the problem as when I revert to your version and change the escape for the `&` char to `'&ampssssss;'` The html exported still use `&amp;` which is quite wierd and I'll re-investigate the problem.